### DISCUSSION TITLE
guide?

### DISCUSSION THREAD

**Discussion by typelumina** [DaZombieKiller/FatePackageManager#6] [Q&A]
i'm sorry, i'm new to decrypting game files and i was wondering if there was some sort of "read me" or steps to follow on how to use this package manager, cause i'm struggling to. thank you for your time in advance, i hope this isn't too troubling

**Comment by DaZombieKiller** [comment]
It's a command-line tool and it has no UI currently, so it may be a bit confusing to work with. You will need to compile it, as there are no pre-compiled releases available yet.

1. Install the [.NET 8 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-8.0.303-windows-x64-installer)
2. Run `publish.cmd`
3. Assuming it worked, you will have an `fpm.exe` in `artifacts/publish/FatePackageManager/release`

To extract a package file, you can just drag it onto `fpm.exe` which will extract it into the same directory. Alternatively, on the command line (i.e. using Command Prompt, PowerShell, etc) you can specify a destination: `fpm.exe input_package_file.dat path/to/output/folder`

Repacking works similarly: `fpm.exe pack path/to/input/folder output_package_file.dat`

**Comment by typelumina** [comment]
thank you, this worked! if i may ask, is the script of the game in the .epk files? since that's what i'm looking to work with. i noticed there are two separate folders with epk files (the main epk one and one under "locale > us") and then also .tl files. thank you again

**Comment by DaZombieKiller** [comment]
The EPK files are essentially string databases, they contain the actual text used by the scripts (the .KS files). They are encrypted separately and require different tools to work with. There are some tools being worked on that make the process easy but they aren't public yet. kurikomoe has also released their own set of tools for EPK files that you can find [here](https://github.com/kurikomoe/FSNr_tools).
